API_Products/controllers/publisher_controller.py

```diff
@@ -2,29 +2,43 @@
 
 
 def insert_publisher(dict_values: dict) -> dict:
-    # Checks the body request:
+    """
+    Método para cadastrar nova editora.
+
+    Verifica se os dados enviados são válidos. Retorna dados de erro caso estejam inválidos;
+    Verifica se a editora já está cadastrada. Retorna dados de erro caso já esteja;
+    Verifica se o país está cadastrado. Retorna dados de erro caso não esteja;
+    Salva os dados da editora. Retorna mensagem de sucesso.
+    Caso ocorra algum erro ao salvar, retorna dados de erro.
+
+    :param dict_values: {name, country}
+    :return: dict or raise Exception
+    """
     received = set(dict_values.keys())
     expected = {"name", "country"}
-    dict_response = dict(status=400, error="Valores inseridos inválidos.", message="Verifique os dados informados.")
-    if not expected.issubset(received):
-        return dict_response
-    elif len(dict_values["name"]) == 0 or len(dict_values["country"]) == 0:
-        return dict_response
+    if expected.symmetric_difference(received) or not dict_values["name"] or not dict_values["country"]:
```

Comment:
Sugestão:
Trocar **expected.symmetric_difference(received) or not dict_values["name"] or not dict_values["country"]**
Por: **received != expected**
Que da o mesmo resultado.

Reply:
Feito